### DISCUSSION TITLE
Add Cover ventilation position

### DISCUSSION THREAD

**Discussion by p0l0** [home-assistant/architecture#502] [Entity Models]
## Context

My garage door (Hörmann, one of the most used brands in Germany) has a "ventilation" position and I think many others have it also. The garage door doesn't offer the possibility to set a percentage, so you are currently only able to open, stop or close the door. But the door offers a direct option to put it to a preconfigured "ventilation" position.

Currently the only way to get automatically into the "ventilation" position, is to trigger the call open the door and to stop after "x" seconds.

## Proposal

As garage door is handled as cover, we should extend the integration to handle the ventilation position.

We should add the state "ventilation" and a service "set_ventilation" to the cover integration.

A proposal for the fronted would be: 

![cover-set-ventilation](https://user-images.githubusercontent.com/1452110/104587311-0dea2100-5667-11eb-820b-5309179a97eb.png)

## Consequences

It wouldn't have any effect for current covers which doesn't support this feature. The only part to be considered is how to display it in the frontend, without changing too much the current layout.

If the cover/garage door does support it, you will be able to change directly to ventilation position, without needing to create complicated automation based on counting the seconds until reaching the ventilation position.

**Comment by frenck** [comment]
In general, we only want to add things to the entity model, if multiple brand/integrations (could) support such a thing. If not, it, in general, should be a service specifically for the integration that provides this functionality.

Could you provide more examples of brands/integrations in Home Assistant that support this feature to justify the addition of suggested changes?

**Comment by p0l0** [comment]
After some research, I found that door operators from SOMMER using SOMWeb (https://github.com/taarskog/pysomweb) also support ventilation, but it's not implemented in the library. The same for the door operators from Ra­de­ma­cher, which uses DuoFern (https://github.com/gluap/pyduofern).

I'm using the Homematic IP Module for Hörmann drives (HmIP-MOD-HO), but it's also possible to use the Hörmann KNX Gateway.

The door openers from Novoferm, also are able to set ventilation position, using the HomematicIP Module (HmIP-MOD-TM).

I think it's a common usage for many brands, because combined with a humidity sensor, you can automatically ventilate the Garage. I have seen that other brands also support it, but they currently only offer it in combination with an own humidity sensor and without any API.

**Comment by MaximilianMM** [comment]
I'm also very interested in this functionality. Is there somebody who has achieved this with a custom implementation?

**Comment by CrEaK** [comment]
Im currently moving to HomeAssistant and i also missing this feature for my Homematic Device (HmIP-MOD-HO). Maybe I will try by myself and change to code to integrate an seperate switch for the ventilation, but as far as I know there is no "ventilation" status on the cover-type...

**Comment by Xylaw** [comment]
I am also very interested in this functionality. I use the Homematic IP device HmIP-MOD-TM in combination with a Novomatic N563S garage door opener from Novoferm.

**Reply by Boarder** [reply]
Hi have the TM Modul und would be very interested as well. The wireless garage door opener has the function and under the tools entities i also see a ventilation option but do not know how to use it.
